react-events-and-state/src/index.jsx

```diff
@@ -1,57 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ReactDOM from 'react-dom';
 
-class CustomButton extends React.Component {
-  // FIRST instantiate a CustomButton object from this class
-  constructor(props) {
-    super(props);
-    this.state = { isClicked: false };
+function CustomButton() {
+  const [buttonTxt, setText] = useState(() => {
+    return 'Click Me!';
+  });
+  const [clicked, setClick] = useState(() => {
+    return false;
+  });
```

Comment:
Since `clicked` ultimately determines the button text, avoid having a state for it.